feature/login/src/main/java/com/epmedu/animeal/login/domain/EnterCodeViewModel.kt

```diff
@@ -0,0 +1,97 @@
+package com.epmedu.animeal.login.domain
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.epmedu.animeal.login.domain.model.EnterCodeScreenModel
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+
+internal class EnterCodeViewModel : ViewModel() {
+
+    private val _model = MutableStateFlow(
+        EnterCodeScreenModel(
+            phoneNumber = getPhoneNumber(),
+            code = getEmptyCode(),
+        )
+    )
+    val model: StateFlow<EnterCodeScreenModel> = _model.asStateFlow()
+    private val _isCodeCorrect = MutableStateFlow(false)
+    val isCodeCorrect: StateFlow<Boolean> get() = _isCodeCorrect.asStateFlow()
+
+    private val currentModel get() = _model.value
+
+    init {
+        viewModelScope.launch {
+            launchResendTimer()
+        }
+    }
+
+    private fun getPhoneNumber(): String {
+        // TODO: Retrieve a phone number from repository
+        return PHONE_NUMBER_PLACEHOLDER
+    }
+
+    private fun getEmptyCode() = List(CODE_SIZE) { null }
+
+    private suspend fun launchResendTimer() {
+        for (tick in RESEND_DELAY downTo 1L) {
+            _model.emit(currentModel.copy(resendDelay = tick))
```

Comment:
As you use _StateFlow_, not _SharedFlow_, I think you can just set _value_ instead of emitting it.
e.g. `_model.value = currentModel.copy(resendDelay = tick)`

Reply:
Good point. Changed it, thanks.